source/fundamentals/aggregation.txt

```diff
@@ -0,0 +1,221 @@
+
+.. note::
+  
+  This example uses the ``sample_restaurants.restaurants`` collection
+  from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+  free MongoDB Atlas cluster and load the sample datasets, see :ref:`csharp-quickstart`.
+
+The following code example produces a count of the number of bakeries in each borough
+of New York. To do so, it uses an aggregation pipeline with the following stages:
+
+- A :manual:`$match </reference/operator/aggregation/match/>` stage to filter for documents whose
+  ``cuisine`` field contains the value ``"Bakery"``.
+
+- A :manual:`$group </reference/operator/aggregation/group/>` stage to group the matching
+  documents by the ``borough`` field, accumulating a count of documents for each distinct value
+  of that field.
+
+The following sections contain versions of this example that each provide different 
+approaches to creating and combining the aggregation stages used in the example pipeline. 
+
+LINQ Approach
+~~~~~~~~~~~~~
+
+.. io-code-block::
+
+   .. input:: /includes/fundamentals/code-examples/LinqAggregation.cs
+      :language: csharp
+      :dedent:
+      :start-after: begin-aggregation
+      :end-before: end-aggregation
+   
+   .. output::
+      :language: console
+      :visible: false
+
+      { _id = Bronx, Count = 71 }
+      { _id = Brooklyn, Count = 173 }
+      { _id = Staten Island, Count = 20 }
+      { _id = Missing, Count = 2 }
+      { _id = Manhattan, Count = 221 }
+      { _id = Queens, Count = 204 }
+
+To learn more about using LINQ to construct aggregation pipelines, see the
+:ref:`csharp-linq` guide.      
+
+Builders Approach
+~~~~~~~~~~~~~~~~~
+
+.. io-code-block::
+
+   .. input:: /includes/fundamentals/code-examples/BuilderAggregation.cs
+      :language: csharp
+      :dedent:
+      :start-after: begin-aggregation
+      :end-before: end-aggregation
+   
+   .. output::
+      :language: console
+      :visible: false
+
+      { _id = Bronx, Count = 71 }
+      { _id = Brooklyn, Count = 173 }
+      { _id = Staten Island, Count = 20 }
+      { _id = Missing, Count = 2 }
+      { _id = Manhattan, Count = 221 }
+      { _id = Queens, Count = 204 }
+
+To learn more about using builders to construct aggregation pipelines,
+see the :ref:`csharp-builders-aggregation` section of the Operations with Builders guide.       
+
+BsonDocument Approach
+~~~~~~~~~~~~~~~~~~~~~
+
+.. io-code-block::
+
+   .. input:: /includes/fundamentals/code-examples/Aggregation.cs
+      :language: csharp
+      :dedent:
+      :start-after: begin-aggregation
+      :end-before: end-aggregation
+
+   .. output::
+      :language: console
+      :visible: false
+
+      { "_id" : "Brooklyn", "count" : 173 }
+      { "_id" : "Manhattan", "count" : 221 }
+      { "_id" : "Bronx", "count" : 71 }
+      { "_id" : "Missing", "count" : 2 }
+      { "_id" : "Staten Island", "count" : 20 }
+      { "_id" : "Queens", "count" : 204 }
+
```

Comment:
S: add a note here explaining why users should prefer to use one of the other APIs